build.sbt

```diff
@@ -34,24 +34,53 @@ addCommandAlias("check", "all scalafmtSbtCheck scalafmtCheck test:scalafmtCheck"
 val zioVersion       = "1.0.0-RC17"
 val zioNioVersion    = "1.0.0-RC2"
 val zioConfigVersion = "1.0.0-RC10"
-libraryDependencies ++= Seq(
-  "dev.zio"        %% "zio"                 % zioVersion,
-  "dev.zio"        %% "zio-test"            % zioVersion % "test",
-  "dev.zio"        %% "zio-test-sbt"        % zioVersion % "test",
-  "dev.zio"        %% "zio-nio"             % zioNioVersion,
-  "dev.zio"        %% "zio-config-typesafe" % zioConfigVersion,
-  "org.scala-lang" % "scala-reflect"        % scalaVersion.value
-)
-
-testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework"))
 
 lazy val root =
-  (project in file("."))
+  project
+    .in(file("."))
+    .settings(skip in publish := true)
+    .aggregate(zioActors, zioActorsPersistence)
+
+lazy val zioActors = module("zio-actors", "actors")
+  .enablePlugins(BuildInfoPlugin)
+  .settings(buildInfoSettings("zio.actors"))
+  .settings(
+    libraryDependencies ++= Seq(
+      "dev.zio"        %% "zio"                 % zioVersion,
+      "dev.zio"        %% "zio-test"            % zioVersion % "test",
+      "dev.zio"        %% "zio-test-sbt"        % zioVersion % "test",
+      "dev.zio"        %% "zio-nio"             % zioNioVersion,
+      "dev.zio"        %% "zio-config-typesafe" % zioConfigVersion,
+      "org.scala-lang" % "scala-reflect"        % scalaVersion.value
+    ),
+    testFrameworks := Seq(new TestFramework("zio.test.sbt.ZTestFramework"))
+  )
+  .settings(
+    stdSettings("zio-actors")
+  )
+
+lazy val zioActorsPersistence = module("zio-actors-persistence", "persistence")
```

Comment:
I'm not sure this should be "the" persistence module, as we might wanna be backed by various storages. If possible, extract the postgres-specific stuff into the separate module.

Reply:
Sure, so there will be a `zio-actors-persistence` with just in-mem and `zio-actors-persistence-jdbc`.

My question is: As the only thing that determines plugin usage (e.g. jdbc plugin) is a dependency in `build.sbt` and configuration in `application.conf` the core `persistence` module must use reflection to get access to the plugin defined in configuration, right?

Reply:
This is a good question. I'd delegate the decision as much as possible. Core module should have no clue about the concrete bindings, but these bindings must be on path and configured via application.conf.